1.13 announcement

brson opened the pull request.

rust-highfive commented:
r? @nikomatsakis

(rust_highfive has picked a reviewer for you, use r? to override)

brson commented:
[Rendered](https://github.com/brson/blog.rust-lang.org/blob/b07ffa43ea41957cd56fdb2d21751a306d5f965a/_posts/2016-11-09-Rust-1.13.md)

Mark-Simulacrum commented on _posts/2016-11-09-Rust-1.13.md:
````diff
@@ -0,0 +1,475 @@
+foo()?.bar()?.baz()?
+```
+The first is quite difficult to scan visually, and each layer of error handling
+prefixs the expression with an additional call to `try!`. This brings undo
````
nit: undo -> undue

aturon commented on _posts/2016-11-09-Rust-1.13.md:
```diff
@@ -0,0 +1,475 @@
+
+#### Performance improvements
+
+There has bee a lot of focus on compiler performance lately. There's good news
```
s/bee/been/

aturon commented on _posts/2016-11-09-Rust-1.13.md:
```diff
@@ -0,0 +1,475 @@
+* [Implement `Debug` for `DirEntry`][36631]
+* [When `getaddrinfo` returns `EAI_SYSTEM` retrieve actual error from
+  `errno`][36754]
+* [`SipHasher`] is deprecated. Use [`DefaultHasher`].
```
Missing link

aturon commented on _posts/2016-11-09-Rust-1.13.md:
```diff
@@ -0,0 +1,475 @@
+
+#### Cargo features
+
+This release contains critical security updates to Cargo, which depends on
```
I'm a bit worried that this critical security update isn't being flagged high enough in the post -- feels a bit like it's buried? Maybe just forward-reference it?

brson commented:
Thanks for the review.

I've pushed another commit that mentions the security issue at the top, and completely removes the libs and cargo list sections.

My main problems with it now

- There's an unwritten section on type macros and statement attributes. These seem important enough to call out.
- There's no section on diagnostics, where we have some pictures we could use.
- It does not mention the known arm miscompilation

aturon commented:
@brson Rather than entirely removing libs and cargo, maybe we can have a section "Library stabilizations and Cargo features" that just mentions there were a number of stabilizations and features, and points to the relnotes for details?

One thing I forgot to mention: the inclusion of LLVM IR was kind of... meh... to me. It feels too insider baseball.

> It does not mention the known arm miscompilation

This seems important, though embarrassing. One saving grace is it's a tier 2 platform...

> The graph, while providing something to look at, doesn't look as compelling as it could, with lots of flat data.

I think it's fine.

lqd commented on _posts/2016-11-09-Rust-1.13.md:
```diff
@@ -0,0 +1,375 @@
+here we are handling errors in the standard way, by passing them up the
+call stack.
+
+Seasoned Rustaceans may recongize that this is the same as the `try!` macro
```
tiny typo: "recongize"

steveklabnik commented on _posts/2016-11-09-Rust-1.13.md:
```diff
@@ -0,0 +1,375 @@
+conferences in short succession, and all the while we've been hacking
+away. There's a whole bunch of amazing stuff coming down the pipeline, but in
+today's release we're highlighting three new language features, including the
+long-awaited `?` operator, as well as a surge in progress on compile-time
```
Talking about a "surge in progress" is complicated here, because you want to talk about progress increasing on making something decrease, which sounds weird. The first time I read this, I thought this meant that compile times increased.

Maybe "as well as significant speed improvements in compile times."

steveklabnik commented on _posts/2016-11-09-Rust-1.13.md:
```diff
@@ -0,0 +1,375 @@
+performance.
+
+This release contains important security updates to Cargo, which depends on curl
+and OpenSSL, which both published critical security updates recently. For more
```
People are going to ask "why wasn't this released immediately" and I don't know the answer to that either

brson replied:
The curl release was only last week, so it was too late to put out a 1.12 patch. OpenSSL I don't recall.

steveklabnik commented on _posts/2016-11-09-Rust-1.13.md:
````diff
@@ -0,0 +1,375 @@
+succession. Consider:
+
+```rust
+try!(try!(try!(foo()).bar()).baz())
````
people asked in my PR why I did it this way, it's because it was from the justification in the RFC. If nobody writes code this way, then the RFC is poorly justified 😉

steveklabnik commented on _posts/2016-11-09-Rust-1.13.md:
````diff
@@ -0,0 +1,375 @@
+```rust
+foo()?.bar()?.baz()?
+```
+The first is quite difficult to scan visually, and each layer of error handling
````
i'd put a blank line above this one

brson commented:
> It feels too insider baseball.

I removed it. Although this was kinda meh, I do think our technical audience would care about interesting implementation details.

Fixed a few edits. I'll add another section talking about cargo and libs and mention the arm errata. A section on the two other features would still be desirable.

steveklabnik commented:
One last issue: this file has the wrong name, sorry. it should be 11-10 not 11-09. this will affect the URL

brson commented:
Date fixed. More sections added. [Rendered](https://github.com/brson/blog.rust-lang.org/blob/1.13-announcement/_posts/2016-11-10-Rust-1.13.md).

Manishearth commented:
A bit on inline drop https://github.com/Manishearth/blog.rust-lang.org/commit/f72bb0b582137824371f4c2dd2c2baea6af4fdd6
